### ISSUE TITLE
Spin-polarized quantities return order

### ISSUE THREAD

**Issue by zerothi** [zerothi/sisl#501]
**Describe the issue**

#500 brought about this issue. (@tfrederiksen )

The problem is the handling of the spin-densities is that the current PDOS (for polarized) returns [up, down], whereas NC/SOC returns [total, x, y, z].

This issue is to discuss the option to change the polarized return order to be [total, z].

POSITIVE:
- this would be conforming to the NC/SOC calculations
- the code would always be the same if users just wanted to plot the total charge (`[0]`)
SIESTA:
- Return values would not be the same as the output data (say Mulliken charges)

Comments would be much appreciated!

**Comment by zerothi** [comment]
Oh, and should the entries be `no, spin` or `spin, no`? While we are at it!

**Comment by tfrederiksen** [comment]
I am used to quantities with `(up, down)` components.

**Comment by zerothi** [comment]
> I am used to quantities with `(up, down)` components.

what if you then do a NC calculation, then the components would be *very* different?

@pfebrer @ahkole @juijan you seem to do more SOC calculations, what are your comments?

**Comment by pfebrer** [comment]
I have never done a (meaningful) SOC calculation :)

From my experience handling NC spins in the visualization module, I agree that it is a bit of a pain to have a completely different way of processing NC and diagonal spins.

But from my experience as a user, up/down components feel very intuitive. It is not easy for me to understand what `total` and `z` would store and how would you convert to `up`/`down`. Could you perhaps clarify it with an example of an array storing eigenvalues for a given k point?

**Comment by zerothi** [comment]
Total = up + down
z = up - down

So for ltotal, z] we can calculate:
up = (total + z)/2
down = (total - z)/2

**Comment by pfebrer** [comment]
(I'm thinking about eigenstate energies in particular)

**Comment by zerothi** [comment]
sure they do. `total` is the total charge. `z` is the actual spin of the system.

**Comment by pfebrer** [comment]
Aaaah ok, this is specific for densities, not in general for any quantity.

**Comment by juijan** [comment]
I would prefer [total, z] for consistency. 

Typically the spin and the total charge are what is most relevant, in my opinion, not what the charge in either spin channel is on its own.

Regarding the index order, I would say first spin then orbitals.

**Comment by ahkole** [comment]
> I would prefer [total, z] for consistency.

I would agree with this. I think I use the total charge most often and I feel like [total, z] are pretty intuitive physical quantities.

> Regarding the index order, I would say first spin then orbitals.

Also agree with this. For me the most intuitive way to sequentially write the data would be first all the densities for one spin component, then for the next spin component, etc.. And considering that numpy arrays are by default in row-major order this would correspond with the `spin, no` order of the indices.

**Comment by zerothi** [comment]
Ok, I think we will update arrays to be
```python
spin, no
```
And store spin polarized as `total, z`. It seems to give the most transparent usage when doing both polarized and non collinear calculations.

**Comment by zerothi** [comment]
What about spin-moment calculations? Should they also be returning `[spin, no]`?

**Comment by zerothi** [comment]
I think, for consistency, everything will be `spin, no`
